src/famly_fetch/downloader.py

```diff
@@ -59,18 +67,32 @@ def download_images_from_notes(self, child_id, first_name):
 
         while True:
             click.echo("Fetching next 100 notes")
-            batch = self._apiClient.get_child_notes(child_id, next=next_ref, first=100)
+            batch = self._apiClient.get_child_notes(
+                child_id, cursor=next_ref, first=100
+            )
             click.echo(f"{len(batch['result'])} fetched.")
 
             for _i, note in enumerate(batch["result"]):
                 text = note["text"] + " - " + note["createdBy"]["name"]["fullName"]
                 date = note["createdAt"]
 
-                for img in note["images"]:
-                    click.echo(f" - image {img['id']} from note at {date}")
-                    url = self.get_secret_image_url(img)
-
-                    self.fetch_image(url, img["id"], f"{first_name}-note", date, text)
+                for img_dict in note["images"]:
+                    img = SecretImage.from_dict(
+                        img_dict, date_override=date, text_override=text
+                    )
+                    click.echo(f" - image {img.img_id} from note at {img.date}")
+
+                    file_path = self.download_file_path(img, f"{first_name}-note")
+                    if file_path.is_file and file_path.exists():
+                        click.secho(
+                            f"File {file_path} already exists, {'stopping download' if self.stop_on_existing else 'skipping'}.",
+                            fg="yellow",
+                        )
+                        if self.stop_on_existing:
+                            return
+                        else:
+                            continue
```

Comment:
main logic that stop the download when it's encountering a file which is already existed in the download folder.